tasks/ast_tools/src/derives/content_eq.rs

```diff
@@ -0,0 +1,112 @@
+use itertools::Itertools;
+use proc_macro2::TokenStream;
+use quote::quote;
+
+use crate::{
+    codegen::LateCtx,
+    schema::{EnumDef, GetGenerics, StructDef, ToType, TypeDef},
+    util::ToIdent,
+};
+
+use super::{define_derive, Derive, DeriveOutput};
+
+define_derive! {
+    pub struct DeriveContentEq;
+}
+
+const IGNORE_FIELDS: [(/* field name */ &str, /* field type */ &str); 5] = [
+    ("span", "Span"),
+    ("trailing_comma", "Span"),
+    ("scope_id", "ScopeId"),
+    ("symbol_id", "SymbolId"),
+    ("reference_id", "ReferenceId"),
+];
```

Comment:
Could we make this a list of types to ignore, rather than field name + type pairs?

I assume we should be ignoring *any* `Span` (regardless of what field it appears in). If we add any new fields which contain a `Span` to AST types in future, we'll have to remember to add them here too.

Boshen had to remember to do that in his amends to #5327 (and he did remember), but if this PR wasn't fresh in the mind, it would have been very easy to forget.

Reply:
I wasn't sure about it since our types aren't context-aware so for less common type names you can have code like this:

```rust
// locally used type name.
type ReferenceId = usize;
#[ast]
#[generate_derive(ContentEq)]
struct X {
    span: Span,
    foo: ReferenceId,
}
```

Something like `ReferenceId` isn't that crazy to be used by someone who isn't aware of this and would mess up their equality checks for good(it is always true in this example).

What if we ignore the `Span` type name but use the `name + type` combo for the rest? Or we can actually look at the use statements and resolve the type names, I think it would also be a welcomed change in the `oxc_regular_expression` crate, `Atom` in regex means something else, and our atom is usually used there as `use oxc_span::Atom as SpanAtom`, But I had to change it back to `Atom` in the ast file to make the codegen recognize it.